teamcity: failed tests on master: testrace/TestParallel/subquery_retry_multinode, testrace/TestParallel/subquery_retry, testrace/TestParallel

The following tests appear to have failed:

[#220991](https://teamcity.cockroachdb.com/viewLog.html?buildId=220991):

```
--- FAIL: testrace/TestParallel/subquery_retry_multinode (0.000s)
Test ended in panic.

------- Stdout: -------
E170413 18:05:49.407690 322006 storage/replica_proposal.go:458  [n1,s1,r1/1:/M{in-ax},@c425cd9200] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
E170413 18:05:52.322987 323192 storage/queue.go:634  [split,n5,s5,r2/3:/System/{""-tsd},@c4200c5800] unexpected value: raw_bytes:"\204\207\034.\003\010\002\022\001\004\032\004\004tsd\"\006\010\003\020\003\030\004\"\006\010\004\020\004\030\002\"\006\010\005\020\005\030\003\"\006\010\002\020\002\030\005(\006" timestamp:<wall_time:1492106752133944352 logical:0 > 
--- done: testdata/parallel_test/subquery_retry_multinode/setup: 2 tests, 0 failures
--- done: testdata/parallel_test/subquery_retry_multinode/txn: 1 tests, 0 failures
--- done: testdata/parallel_test/subquery_retry_multinode/txn: 1 tests, 0 failures
--- done: testdata/parallel_test/subquery_retry_multinode/txn: 1 tests, 0 failures
--- done: testdata/parallel_test/subquery_retry_multinode/txn: 1 tests, 0 failures
panic: test timed out after 15m0s

goroutine 339361 [running]:
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1023 +0x139
created by time.goFunc
	/usr/local/go/src/time/sleep.go:170 +0x52

goroutine 1 [chan receive, 11 minutes]:
testing.(*T).Run(0xc4201b2410, 0x241a55b, 0xc, 0x248e670, 0xc4204c5b01)
	/usr/local/go/src/testing/testing.go:698 +0x582
testing.runTests.func1(0xc4201b2410)
	/usr/local/go/src/testing/testing.go:882 +0xab
testing.tRunner(0xc4201b2410, 0xc4204c5d10)
	/usr/local/go/src/testing/testing.go:657 +0x108
testing.runTests(0xc42010dec0, 0x32cbf80, 0x88, 0x88, 0x2f6afbd)
	/usr/local/go/src/testing/testing.go:888 +0x4e1
testing.(*M).Run(0xc420273f28, 0x40e8140)
	/usr/local/go/src/testing/testing.go:822 +0x1c4
github.com/cockroachdb/cockroach/pkg/sql_test.TestMain(0xc4204c5f28)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:227 +0x1b7
main.main()
	github.com/cockroachdb/cockroach/pkg/sql/_test/_testmain.go:340 +0x210

goroutine 17 [syscall, 15 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1

goroutine 19 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).flushDaemon(0x3721f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1022 +0x85
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:583 +0x120

goroutine 20 [chan receive, 11 minutes]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0x3721f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1049 +0x82
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:584 +0x144

goroutine 322558 [chan receive, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func6()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:528 +0xd2
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc42519a980)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322353 [IO wait]:
net.runtime_pollWait(0x7fea148e4ec8, 0x72, 0x72d2da)
	/usr/local/go/src/runtime/netpoll.go:164 +0x5e
net.(*pollDesc).wait(0xc42676fcd8, 0x72, 0x32d0e08, 0xffffffffffffffff)
	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x5b
net.(*pollDesc).waitRead(0xc42676fcd8, 0xc4232a4000, 0x8000)
	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x42
net.(*netFD).Read(0xc42676fc70, 0xc4232a4000, 0x8000, 0x8000, 0x0, 0x32db6c0, 0x32d0e08)
	/usr/local/go/src/net/fd_unix.go:250 +0x22b
net.(*conn).Read(0xc420cba6c8, 0xc4232a4000, 0x8000, 0x8000, 0xcf8431, 0x0, 0x24907c8)
	/usr/local/go/src/net/net.go:181 +0x97
net.(*TCPConn).Read(0xc420cba6c8, 0xc4232a4000, 0x8000, 0x8000, 0x1, 0x32d31c0, 0xc420074040)
	<autogenerated>:72 +0x73
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*MuxConn).Read(0xc427e07360, 0xc4232a4000, 0x8000, 0x8000, 0xc422b154d0, 0xa4ad56, 0xc422b154d0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux/cmux.go:218 +0x183
crypto/tls.(*block).readFromUntil(0xc426344e70, 0x7fea78e218c8, 0xc427e07360, 0x5, 0xc427e07360, 0xc42a41caf0)
	/usr/local/go/src/crypto/tls/conn.go:488 +0x114
crypto/tls.(*Conn).readRecord(0xc429029180, 0x2490a17, 0xc4290292a0, 0x6cbd4a)
	/usr/local/go/src/crypto/tls/conn.go:590 +0x13d
crypto/tls.(*Conn).Read(0xc429029180, 0xc421172000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1134 +0x190
bufio.(*Reader).Read(0xc429391140, 0xc4278b61f8, 0x9, 0x9, 0x4de1128, 0xc4273cef98, 0x0)
	/usr/local/go/src/bufio/bufio.go:213 +0x7e7
io.ReadAtLeast(0x32d2ac0, 0xc429391140, 0xc4278b61f8, 0x9, 0x9, 0x9, 0xc422b15a48, 0x71e4a3, 0xc42029c340)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d2ac0, 0xc429391140, 0xc4278b61f8, 0x9, 0x9, 0xc4290b0050, 0x0, 0x1)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.readFrameHeader(0xc4278b61f8, 0x9, 0x9, 0x32d2ac0, 0xc429391140, 0x0, 0xc400000000, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:237 +0xa3
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.(*Framer).ReadFrame(0xc4278b61c0, 0x0, 0x32d31c0, 0xc420074040, 0xc42939ce00)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:492 +0x118
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*framer).readFrame(0xc42479c5d0, 0xc4273cf080, 0xc4273cf080, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http_util.go:508 +0x53
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams(0xc4278b6380, 0xc42479c960, 0x248fba0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:307 +0x3c4
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams(0xc42084ce00, 0x32f2040, 0xc4278b6380)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:505 +0x1e8
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveHTTP2Transport(0xc42084ce00, 0x32f2b20, 0xc429029180, 0x32daac0, 0xc42412a580)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:486 +0x5a6
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn(0xc42084ce00, 0x32f3cc0, 0xc427e07360)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:457 +0x5f4
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).Serve
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:426 +0x69c

goroutine 322963 [select, 1 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startGossip.func4()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1355 +0x68b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc429a10e60, 0xc42c097090)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322836 [select]:
github.com/cockroachdb/cockroach/pkg/gossip.(*client).gossip(0xc42c5595f0, 0x7fea7b8679d0, 0xc42ae52940, 0xc4256ca180, 0x32f2b80, 0xc4243632c0, 0xc429a10e60, 0xc4219a0f30, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/gossip/client.go:325 +0x4f9
github.com/cockroachdb/cockroach/pkg/gossip.(*client).startLocked.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/gossip/client.go:132 +0x6e5
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc429a10e60, 0xc42ae52900)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323931 [IO wait]:
net.runtime_pollWait(0x7fea74a585f0, 0x72, 0x72d2da)
	/usr/local/go/src/runtime/netpoll.go:164 +0x5e
net.(*pollDesc).wait(0xc420ce7cd8, 0x72, 0x32d0e08, 0xffffffffffffffff)
	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x5b
net.(*pollDesc).waitRead(0xc420ce7cd8, 0xc424299800, 0x800)
	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x42
net.(*netFD).Read(0xc420ce7c70, 0xc424299800, 0x800, 0x800, 0x0, 0x32db6c0, 0x32d0e08)
	/usr/local/go/src/net/fd_unix.go:250 +0x22b
net.(*conn).Read(0xc4256128a8, 0xc424299800, 0x800, 0x800, 0x6e, 0x0, 0x24907c8)
	/usr/local/go/src/net/net.go:181 +0x97
net.(*TCPConn).Read(0xc4256128a8, 0xc424299800, 0x800, 0x800, 0x1, 0x32d31c0, 0xc420074040)
	<autogenerated>:72 +0x73
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*MuxConn).Read(0xc4205995e0, 0xc424299800, 0x800, 0x800, 0xc4391674d0, 0xa4ad56, 0xc4391674d0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux/cmux.go:218 +0x183
crypto/tls.(*block).readFromUntil(0xc42390cff0, 0x7fea78e218c8, 0xc4205995e0, 0x5, 0xc4205995e0, 0x6e4aa0)
	/usr/local/go/src/crypto/tls/conn.go:488 +0x114
crypto/tls.(*Conn).readRecord(0xc42b94d500, 0x2490a17, 0xc42b94d620, 0x6cbd4a)
	/usr/local/go/src/crypto/tls/conn.go:590 +0x13d
crypto/tls.(*Conn).Read(0xc42b94d500, 0xc42a288000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1134 +0x190
bufio.(*Reader).Read(0xc421af1b00, 0xc42044bed8, 0x9, 0x9, 0x7fea4850e958, 0xc42b2c57a8, 0x0)
	/usr/local/go/src/bufio/bufio.go:213 +0x7e7
io.ReadAtLeast(0x32d2ac0, 0xc421af1b00, 0xc42044bed8, 0x9, 0x9, 0x9, 0xc439167a48, 0x71e4a3, 0xc420001a00)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d2ac0, 0xc421af1b00, 0xc42044bed8, 0x9, 0x9, 0x0, 0x1c20a50, 0x7fea4850e958)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.readFrameHeader(0xc42044bed8, 0x9, 0x9, 0x32d2ac0, 0xc421af1b00, 0x0, 0xc400000000, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:237 +0xa3
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.(*Framer).ReadFrame(0xc42044bea0, 0x0, 0x32d31c0, 0xc420074040, 0xc421bfb600)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:492 +0x118
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*framer).readFrame(0xc423ba61e0, 0xc42b2c5950, 0xc42b2c5950, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http_util.go:508 +0x53
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams(0xc4210540e0, 0xc423ba6330, 0x248fba0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:307 +0x3c4
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams(0xc42044a9a0, 0x32f2040, 0xc4210540e0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:505 +0x1e8
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveHTTP2Transport(0xc42044a9a0, 0x32f2b20, 0xc42b94d500, 0x32daac0, 0xc4230f9810)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:486 +0x5a6
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn(0xc42044a9a0, 0x32f3cc0, 0xc4205995e0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:457 +0x5f4
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).Serve
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:426 +0x69c

goroutine 323308 [IO wait]:
net.runtime_pollWait(0x7fea7b868550, 0x72, 0x72d2da)
	/usr/local/go/src/runtime/netpoll.go:164 +0x5e
net.(*pollDesc).wait(0xc423df6a78, 0x72, 0x32d0e08, 0xffffffffffffffff)
	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x5b
net.(*pollDesc).waitRead(0xc423df6a78, 0xc420588c00, 0x400)
	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x42
net.(*netFD).Read(0xc423df6a10, 0xc420588c00, 0x400, 0x400, 0x0, 0x32db6c0, 0x32d0e08)
	/usr/local/go/src/net/fd_unix.go:250 +0x22b
net.(*conn).Read(0xc42c572230, 0xc420588c00, 0x400, 0x400, 0x102, 0x6db864, 0x1c33fe0)
	/usr/local/go/src/net/net.go:181 +0x97
net.(*TCPConn).Read(0xc42c572230, 0xc420588c00, 0x400, 0x400, 0xc4221f36e8, 0xa4ad56, 0xc4221f36e8)
	<autogenerated>:72 +0x73
crypto/tls.(*block).readFromUntil(0xc42c144510, 0x7fea78e218a0, 0xc42c572230, 0x5, 0xc42c572230, 0x6e4aa0)
	/usr/local/go/src/crypto/tls/conn.go:488 +0x114
crypto/tls.(*Conn).readRecord(0xc426461500, 0x2490a17, 0xc426461620, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:590 +0x13d
crypto/tls.(*Conn).Read(0xc426461500, 0xc4297f2000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1134 +0x190
bufio.(*Reader).Read(0xc42a4fcf00, 0xc422acd378, 0x9, 0x9, 0x0, 0x0, 0xc4221f3be0)
	/usr/local/go/src/bufio/bufio.go:213 +0x7e7
io.ReadAtLeast(0x32d2ac0, 0xc42a4fcf00, 0xc422acd378, 0x9, 0x9, 0x9, 0xc42029c9c0, 0xffffffff, 0xc420e602d0)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d2ac0, 0xc42a4fcf00, 0xc422acd378, 0x9, 0x9, 0xd6e868, 0xc42af9c2b0, 0x0)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.readFrameHeader(0xc422acd378, 0x9, 0x9, 0x32d2ac0, 0xc42a4fcf00, 0x0, 0xc400000000, 0xc4290bdb00, 0x32d31c0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:237 +0xa3
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.(*Framer).ReadFrame(0xc422acd340, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:492 +0x118
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*framer).readFrame(0xc42044ea80, 0xc42522df20, 0xc42522df00, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http_util.go:508 +0x53
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Client).reader(0xc42af9c180)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:1043 +0x15b
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Client
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:251 +0x1294

goroutine 321955 [chan receive, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:501 +0xad
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc423f12990)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 321958 [chan receive, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*muxListener).Accept(0xc42c379280, 0xc432cb6500, 0xc432cb64e0, 0x32de680, 0xc423f12960)
	<autogenerated>:9 +0x9f
crypto/tls.(*listener).Accept(0xc42c3792e0, 0xc432cb64b0, 0x22489a0, 0x32b2c20, 0x2329d20)
	/usr/local/go/src/crypto/tls/tls.go:52 +0x56
net/http.(*Server).Serve(0xc42acc8a50, 0x32e0f40, 0xc42c3792e0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2643 +0x2a8
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func5()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:524 +0x8f
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc42c379320)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 324039 [select]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Client).keepalive(0xc42c3ed380)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:1162 +0x6e6
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Client
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:283 +0x1a4b

goroutine 322400 [chan receive, 10 minutes]:
database/sql.(*DB).connectionOpener(0xc426e21720)
	/usr/local/go/src/database/sql/sql.go:837 +0x6e
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:582 +0x318

goroutine 322528 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x4b3
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4226e20a0, 0xc4243cf180)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322529 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x4b3
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4226e20a0, 0xc4243cf1c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322784 [IO wait, 10 minutes]:
net.runtime_pollWait(0x7fea7ac253a0, 0x72, 0x32d0e08)
	/usr/local/go/src/runtime/netpoll.go:164 +0x5e
net.(*pollDesc).wait(0xc4227e3808, 0x72, 0x32d0e08, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x5b
net.(*pollDesc).waitRead(0xc4227e3808, 0xffffffffffffffff, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x42
net.(*netFD).accept(0xc4227e37a0, 0x0, 0x32d8940, 0xc42a86f580)
	/usr/local/go/src/net/fd_unix.go:430 +0x2c0
net.(*TCPListener).accept(0xc4230b2188, 0xc42053aaf0, 0xc429142ef8, 0x6cb8a2)
	/usr/local/go/src/net/tcpsock_posix.go:136 +0x51
net.(*TCPListener).Accept(0xc4230b2188, 0x248ee90, 0xc42989d640, 0xc42babca20, 0x6cbb7e)
	/usr/local/go/src/net/tcpsock.go:228 +0x50
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*cMux).Serve(0xc42989d640, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux/cmux.go:124 +0xdd
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:513 +0x47
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc429a10e60, 0xc42bf3a5a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322706 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/sql.(*sqlStats).startResetWorker.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/app_stats.go:248 +0x1ce
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc4292d19b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322846 [select, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Client).controller(0xc42980fc80)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:1116 +0x912
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Client
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:281 +0x1954

goroutine 323250 [chan receive, 10 minutes]:
database/sql.(*DB).connectionOpener(0xc4258b6c80)
	/usr/local/go/src/database/sql/sql.go:837 +0x6e
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:582 +0x318

goroutine 323935 [select]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc420d29cc0, 0xc4221c0d70, 0x5, 0x5, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/transport.go:143 +0x70a
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*Stream).Read(0xc4227af680, 0xc4221c0d70, 0x5, 0x5, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/transport.go:333 +0x7f
io.ReadAtLeast(0x32d7980, 0xc4227af680, 0xc4221c0d70, 0x5, 0x5, 0x5, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d7980, 0xc4227af680, 0xc4221c0d70, 0x5, 0x5, 0x400000003, 0x200000004, 0x8)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*parser).recvMsg(0xc4221c0d60, 0x7fffffff, 0x0, 0x0, 0x0, 0x0, 0x6b320d, 0xc41fac3d16)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/rpc_util.go:243 +0xa0
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.recv(0xc4221c0d60, 0x32ea600, 0x40e86e8, 0xc4227af680, 0x32df800, 0x40e86e8, 0x2342040, 0xc42a785d20, 0x7fffffff, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/rpc_util.go:339 +0x61
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*serverStream).RecvMsg(0xc4212645a0, 0x2342040, 0xc42a785d20, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/stream.go:613 +0x1d5
github.com/cockroachdb/cockroach/pkg/storage.(*multiRaftRaftMessageBatchServer).Recv(0xc420809140, 0x7fea7b7fc800, 0xc42390dfb0, 0xc4232f5080)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft.pb.go:351 +0x86
github.com/cockroachdb/cockroach/pkg/storage.(*RaftTransport).RaftMessageBatch.func1.1.1(0x32f3ba0, 0xc420809140, 0xc4219bab80, 0xc4227af6a8, 0xc42474bfa8)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft_transport.go:300 +0x61
github.com/cockroachdb/cockroach/pkg/storage.(*RaftTransport).RaftMessageBatch.func1.1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft_transport.go:323 +0x61
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc429a10e60, 0xc4201da120)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322619 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/server.(*Node).startComputePeriodicMetrics.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:700 +0x344
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc42c48d340)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 321990 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x4b3
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc42c48c580)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322595 [select, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Client).controller(0xc42495e000)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:1116 +0x912
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Client
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:281 +0x1954

goroutine 322928 [select, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*addrConn).transportMonitor(0xc42353a820)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/clientconn.go:859 +0x7cf
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*ClientConn).resetAddrConn.func1(0xc42353a820)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/clientconn.go:604 +0x267
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*ClientConn).resetAddrConn
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/clientconn.go:605 +0xa4d

goroutine 322531 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x4b3
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4226e20a0, 0xc4243cf200)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322855 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/gossip.(*Gossip).bootstrap.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/gossip/gossip.go:1060 +0x692
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc429a10e60, 0xc425c91fb0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322794 [select, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).controller(0xc426e66000)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:742 +0xa23
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Server
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:161 +0xdf2

goroutine 321991 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x4b3
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc42c48c5a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322848 [select, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*addrConn).transportMonitor(0xc428d32680)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/clientconn.go:859 +0x7cf
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*ClientConn).resetAddrConn
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/clientconn.go:592 +0x9fe

goroutine 322559 [chan receive, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*muxListener).Accept(0xc422167180, 0x248fb68, 0xc42044a000, 0x32f3cc0, 0xc423d568c0)
	<autogenerated>:9 +0x9f
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).Serve(0xc42044a000, 0x32ea000, 0xc422167180, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:396 +0x1f8
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func7()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:539 +0x8d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc4221676e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323999 [select]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Client).keepalive(0xc4219fd680)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:1162 +0x6e6
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Client
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:283 +0x1a4b

goroutine 322821 [IO wait]:
net.runtime_pollWait(0x7fea74a58530, 0x72, 0x72d2da)
	/usr/local/go/src/runtime/netpoll.go:164 +0x5e
net.(*pollDesc).wait(0xc4287e0df8, 0x72, 0x32d0e08, 0xffffffffffffffff)
	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x5b
net.(*pollDesc).waitRead(0xc4287e0df8, 0xc42925e000, 0x4000)
	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x42
net.(*netFD).Read(0xc4287e0d90, 0xc42925e000, 0x4000, 0x4000, 0x0, 0x32db6c0, 0x32d0e08)
	/usr/local/go/src/net/fd_unix.go:250 +0x22b
net.(*conn).Read(0xc42b38c138, 0xc42925e000, 0x4000, 0x4000, 0xcf8431, 0x0, 0x24907c8)
	/usr/local/go/src/net/net.go:181 +0x97
net.(*TCPConn).Read(0xc42b38c138, 0xc42925e000, 0x4000, 0x4000, 0x1, 0x32d31c0, 0xc420074040)
	<autogenerated>:72 +0x73
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*MuxConn).Read(0xc42503b220, 0xc42925e000, 0x4000, 0x4000, 0xc42714d4d0, 0xa4ad56, 0xc42714d4d0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux/cmux.go:218 +0x183
crypto/tls.(*block).readFromUntil(0xc422704ae0, 0x7fea78e218c8, 0xc42503b220, 0x5, 0xc42503b220, 0xc42cfc18f0)
	/usr/local/go/src/crypto/tls/conn.go:488 +0x114
crypto/tls.(*Conn).readRecord(0xc42c3d6700, 0x2490a17, 0xc42c3d6820, 0x6cbd4a)
	/usr/local/go/src/crypto/tls/conn.go:590 +0x13d
crypto/tls.(*Conn).Read(0xc42c3d6700, 0xc422774000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1134 +0x190
bufio.(*Reader).Read(0xc427e7e180, 0xc428555d18, 0x9, 0x9, 0x7fea44609ee8, 0xc422a0ecc8, 0x0)
	/usr/local/go/src/bufio/bufio.go:213 +0x7e7
io.ReadAtLeast(0x32d2ac0, 0xc427e7e180, 0xc428555d18, 0x9, 0x9, 0x9, 0xc42714da48, 0x71e4a3, 0xc420ae0ea0)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d2ac0, 0xc427e7e180, 0xc428555d18, 0x9, 0x9, 0x0, 0x1c20a50, 0x7fea44609ee8)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.readFrameHeader(0xc428555d18, 0x9, 0x9, 0x32d2ac0, 0xc427e7e180, 0x0, 0xc400000000, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:237 +0xa3
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.(*Framer).ReadFrame(0xc428555ce0, 0x0, 0x32d31c0, 0xc420074040, 0xc421bfbfc0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:492 +0x118
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*framer).readFrame(0xc42980a330, 0xc422a0edb0, 0xc422a0edb0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http_util.go:508 +0x53
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams(0xc426e66000, 0xc42980a480, 0x248fba0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:307 +0x3c4
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams(0xc42b4eb6c0, 0x32f2040, 0xc426e66000)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:505 +0x1e8
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveHTTP2Transport(0xc42b4eb6c0, 0x32f2b20, 0xc42c3d6700, 0x32daac0, 0xc4233d3b80)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:486 +0x5a6
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn(0xc42b4eb6c0, 0x32f3cc0, 0xc42503b220)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:457 +0x5f4
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).Serve
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:426 +0x69c

goroutine 322052 [select]:
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).runHeartbeat(0xc429046090, 0xc420b73e80, 0xc4239fbab0, 0xf, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:356 +0xb5b
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).GRPCDial.func1.2.1()
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:297 +0x86
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc432cb6b70)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 321964 [select]:
github.com/cockroachdb/cockroach/pkg/gossip.(*Gossip).manage.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/gossip/gossip.go:1089 +0x4d9
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc4233fe7d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 324101 [select, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc4273ba800, 0xc42ad46fb0, 0x5, 0x5, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/transport.go:143 +0x70a
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*Stream).Read(0xc427e86fc0, 0xc42ad46fb0, 0x5, 0x5, 0x6afa5a, 0xc420476400, 0x30)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/transport.go:333 +0x7f
io.ReadAtLeast(0x32d7980, 0xc427e86fc0, 0xc42ad46fb0, 0x5, 0x5, 0x5, 0x5090abf37b95e960, 0xc42a01bac8, 0x6ac67b)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d7980, 0xc427e86fc0, 0xc42ad46fb0, 0x5, 0x5, 0x13d4e99, 0xc42a01bac8, 0x8)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*parser).recvMsg(0xc42ad46fa0, 0x7fffffff, 0xc42a01bc10, 0x6cbf7b, 0xc42001e600, 0xc42001e600, 0x1, 0x6dbd32)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/rpc_util.go:243 +0xa0
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.recv(0xc42ad46fa0, 0x32ea600, 0x40e86e8, 0xc427e86fc0, 0x32df800, 0x40e86e8, 0x2342120, 0xc421147bc0, 0x7fffffff, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/rpc_util.go:339 +0x61
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*clientStream).RecvMsg(0xc4276662c0, 0x2342120, 0xc421147bc0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/stream.go:386 +0x20e
github.com/cockroachdb/cockroach/pkg/storage.(*multiRaftRaftMessageBatchClient).Recv(0xc42517da40, 0x7fea7b7fc800, 0xc421147b30, 0xc42a945710)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft.pb.go:283 +0x86
github.com/cockroachdb/cockroach/pkg/storage.(*RaftTransport).processQueue.func1.1.1(0x32f3b40, 0xc42517da40, 0xc423a724c0, 0xc424e86200, 0x7fea7b7fc800, 0xc420aa82d0, 0xc4226eff98, 0x6cb8a2)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft_transport.go:467 +0x5a
github.com/cockroachdb/cockroach/pkg/storage.(*RaftTransport).processQueue.func1.1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft_transport.go:484 +0x113
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc42a9456b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323129 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/server.(*Server).startSampleEnvironment.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:901 +0x240
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4249c15e0, 0xc42a86ec60)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323016 [select, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Client).controller(0xc4256cb800)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:1116 +0x912
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Client
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:281 +0x1954

goroutine 322625 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).Start.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:337 +0x19d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc42c48d460)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 321993 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x4b3
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc42c48c5e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322853 [chan receive, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*muxListener).Accept(0xc42bf3a4a0, 0x248ed78, 0xc429a10e60, 0xc4230b2168, 0x32f3cc0)
	<autogenerated>:9 +0x9f
github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith(0xc4230b2168, 0xc429a10e60, 0x32ea000, 0xc42bf3a4a0, 0xc42ad44e10, 0xc42519a1c0, 0xc422084c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:115 +0x61
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func9()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:603 +0x31b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc429a10e60, 0xc42989d6c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322012 [semacquire]:
sync.runtime_notifyListWait(0xc426ec77d0, 0xc40000b955)
	/usr/local/go/src/runtime/sema.go:298 +0x11a
sync.(*Cond).Wait(0xc426ec77c0)
	/usr/local/go/src/sync/cond.go:57 +0x97
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc421937830, 0xc4265768c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xe2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x4b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc424ee5440)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323165 [semacquire]:
sync.runtime_notifyListWait(0xc420875cd0, 0xc40000bba1)
	/usr/local/go/src/runtime/sema.go:298 +0x11a
sync.(*Cond).Wait(0xc420875cc0)
	/usr/local/go/src/sync/cond.go:57 +0x97
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc43b40fa70, 0xc4249c15e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xe2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x4b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4249c15e0, 0xc4282fa1e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323160 [semacquire]:
sync.runtime_notifyListWait(0xc420875cd0, 0xc40000bba2)
	/usr/local/go/src/runtime/sema.go:298 +0x11a
sync.(*Cond).Wait(0xc420875cc0)
	/usr/local/go/src/sync/cond.go:57 +0x97
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc43b40fa70, 0xc4249c15e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xe2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x4b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4249c15e0, 0xc4282fa140)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322008 [semacquire]:
sync.runtime_notifyListWait(0xc426ec77d0, 0xc40000b957)
	/usr/local/go/src/runtime/sema.go:298 +0x11a
sync.(*Cond).Wait(0xc426ec77c0)
	/usr/local/go/src/sync/cond.go:57 +0x97
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc421937830, 0xc4265768c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xe2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x4b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc424ee53c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323317 [IO wait]:
net.runtime_pollWait(0x7fea1b22e970, 0x72, 0x72d2da)
	/usr/local/go/src/runtime/netpoll.go:164 +0x5e
net.(*pollDesc).wait(0xc4257a9178, 0x72, 0x32d0e08, 0xffffffffffffffff)
	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x5b
net.(*pollDesc).waitRead(0xc4257a9178, 0xc422521400, 0x400)
	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x42
net.(*netFD).Read(0xc4257a9110, 0xc422521400, 0x400, 0x400, 0x0, 0x32db6c0, 0x32d0e08)
	/usr/local/go/src/net/fd_unix.go:250 +0x22b
net.(*conn).Read(0xc4230b2300, 0xc422521400, 0x400, 0x400, 0x102, 0x6db864, 0x1c33fe0)
	/usr/local/go/src/net/net.go:181 +0x97
net.(*TCPConn).Read(0xc4230b2300, 0xc422521400, 0x400, 0x400, 0xc422b516e8, 0xa4ad56, 0xc422b516e8)
	<autogenerated>:72 +0x73
crypto/tls.(*block).readFromUntil(0xc429f780f0, 0x7fea78e218a0, 0xc4230b2300, 0x5, 0xc4230b2300, 0x6e4aa0)
	/usr/local/go/src/crypto/tls/conn.go:488 +0x114
crypto/tls.(*Conn).readRecord(0xc42208a700, 0x2490a17, 0xc42208a820, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:590 +0x13d
crypto/tls.(*Conn).Read(0xc42208a700, 0xc42412c000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1134 +0x190
bufio.(*Reader).Read(0xc43b947d40, 0xc428554578, 0x9, 0x9, 0x0, 0x0, 0xc422b51be0)
	/usr/local/go/src/bufio/bufio.go:213 +0x7e7
io.ReadAtLeast(0x32d2ac0, 0xc43b947d40, 0xc428554578, 0x9, 0x9, 0x9, 0xc421360000, 0xffffffff, 0xc424b0bcc0)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d2ac0, 0xc43b947d40, 0xc428554578, 0x9, 0x9, 0xd6e868, 0xc423d87c30, 0x0)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.readFrameHeader(0xc428554578, 0x9, 0x9, 0x32d2ac0, 0xc43b947d40, 0x0, 0xc400000000, 0xc421c258c0, 0x32d31c0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:237 +0xa3
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.(*Framer).ReadFrame(0xc428554540, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:492 +0x118
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*framer).readFrame(0xc420812720, 0xc428a8d470, 0xc428a8d400, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http_util.go:508 +0x53
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Client).reader(0xc423d87b00)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:1043 +0x15b
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Client
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:251 +0x1294

goroutine 323085 [select]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Client).keepalive(0xc42b305980)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:1162 +0x6e6
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Client
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_client.go:283 +0x1a4b

goroutine 322991 [select, 10 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).controller(0xc4245c7ea0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:742 +0xa23
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.newHTTP2Server
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:161 +0xdf2

goroutine 322563 [select]:
github.com/cockroachdb/cockroach/pkg/gossip.(*Gossip).manage.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/gossip/gossip.go:1089 +0x4d9
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc426b21880)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 321949 [chan receive, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/rpc.NewContext.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:176 +0x95
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4265768c0, 0xc42c378720)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 324041 [select]:
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).runHeartbeat(0xc42384dd40, 0xc428112fc0, 0xc425c918d0, 0xf, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:356 +0xb5b
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).GRPCDial.func1.2.1()
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:297 +0x86
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc429a10e60, 0xc42280e630)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323158 [semacquire]:
sync.runtime_notifyListWait(0xc420875cd0, 0xc40000bb9b)
	/usr/local/go/src/runtime/sema.go:298 +0x11a
sync.(*Cond).Wait(0xc420875cc0)
	/usr/local/go/src/sync/cond.go:57 +0x97
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc43b40fa70, 0xc4249c15e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xe2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x4b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4249c15e0, 0xc4282fa100)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322416 [chan receive, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func6()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:528 +0xd2
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4226e20a0, 0xc42af15d00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 323989 [select]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc420550680, 0xc423665910, 0x5, 0x5, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/transport.go:143 +0x70a
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*Stream).Read(0xc425945b00, 0xc423665910, 0x5, 0x5, 0x1c32ae0, 0x7fea4853ebf8, 0xc421d840a8)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/transport.go:333 +0x7f
io.ReadAtLeast(0x32d7980, 0xc425945b00, 0xc423665910, 0x5, 0x5, 0x5, 0xc421d840a8, 0x6ffc7c, 0x71e4a3)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d7980, 0xc425945b00, 0xc423665910, 0x5, 0x5, 0xc421d840a8, 0x4, 0x6ffb62)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*parser).recvMsg(0xc423665900, 0x7fffffff, 0x35, 0x35, 0xc4215fbd48, 0x6cbf7b, 0x6b320d, 0xc41fd23350)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/rpc_util.go:243 +0xa0
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.recv(0xc423665900, 0x32ea600, 0x40e86e8, 0xc425945b00, 0x32df800, 0x40e86e8, 0x2342040, 0xc425b995e0, 0x7fffffff, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/rpc_util.go:339 +0x61
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*serverStream).RecvMsg(0xc42bfcf540, 0x2342040, 0xc425b995e0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/stream.go:613 +0x1d5
github.com/cockroachdb/cockroach/pkg/storage.(*multiRaftRaftMessageBatchServer).Recv(0xc420809920, 0x7fea7b7fc800, 0xc4201dbd70, 0xc420801e00)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft.pb.go:351 +0x86
github.com/cockroachdb/cockroach/pkg/storage.(*RaftTransport).RaftMessageBatch.func1.1.1(0x32f3ba0, 0xc420809920, 0xc424e86200, 0xc425945b28, 0xc42474bfa8)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft_transport.go:300 +0x61
github.com/cockroachdb/cockroach/pkg/storage.(*RaftTransport).RaftMessageBatch.func1.1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft_transport.go:323 +0x61
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc4201dbda0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322954 [semacquire]:
sync.runtime_notifyListWait(0xc42989d890, 0xc40000b756)
	/usr/local/go/src/runtime/sema.go:298 +0x11a
sync.(*Cond).Wait(0xc42989d880)
	/usr/local/go/src/sync/cond.go:57 +0x97
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42384ddd0, 0xc429a10e60)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xe2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x4b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc429a10e60, 0xc422dd40c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322582 [select]:
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).runHeartbeat(0xc421d84000, 0xc42b95f640, 0xc426b21780, 0xf, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:356 +0xb5b
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).GRPCDial.func1.2.1()
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:297 +0x86
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc426e217c0, 0xc421147dd0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x8b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x74

goroutine 322443 [IO wait]:
net.runtime_pollWait(0x7fea7b8689d0, 0x72, 0x72d2da)
	/usr/local/go/src/runtime/netpoll.go:164 +0x5e
net.(*pollDesc).wait(0xc421bad1e8, 0x72, 0x32d0e08, 0xffffffffffffffff)
	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x5b
net.(*pollDesc).waitRead(0xc421bad1e8, 0xc420478400, 0x400)
	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x42
net.(*netFD).Read(0xc421bad180, 0xc420478400, 0x400, 0x400, 0x0, 0x32db6c0, 0x32d0e08)
	/usr/local/go/src/net/fd_unix.go:250 +0x22b
net.(*conn).Read(0xc423988950, 0xc420478400, 0x400, 0x400, 0x69, 0x0, 0x24907c8)
	/usr/local/go/src/net/net.go:181 +0x97
net.(*TCPConn).Read(0xc423988950, 0xc420478400, 0x400, 0x400, 0x1, 0x32d31c0, 0xc420074040)
	<autogenerated>:72 +0x73
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*MuxConn).Read(0xc4258b6f00, 0xc420478400, 0x400, 0x400, 0xc42a5e94d0, 0xa4ad56, 0xc42a5e94d0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux/cmux.go:218 +0x183
crypto/tls.(*block).readFromUntil(0xc4268ee180, 0x7fea78e218c8, 0xc4258b6f00, 0x5, 0xc4258b6f00, 0x6e4aa0)
	/usr/local/go/src/crypto/tls/conn.go:488 +0x114
crypto/tls.(*Conn).readRecord(0xc42403fc00, 0x2490a17, 0xc42403fd20, 0x6cbd4a)
	/usr/local/go/src/crypto/tls/conn.go:590 +0x13d
crypto/tls.(*Conn).Read(0xc42403fc00, 0xc424684000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1134 +0x190
bufio.(*Reader).Read(0xc427408300, 0xc426184118, 0x9, 0x9, 0x7fea24014f78, 0xc4273cf1d8, 0x0)
	/usr/local/go/src/bufio/bufio.go:213 +0x7e7
io.ReadAtLeast(0x32d2ac0, 0xc427408300, 0xc426184118, 0x9, 0x9, 0x9, 0xc42a5e9a48, 0x71e4a3, 0xc420912ea0)
	/usr/local/go/src/io/io.go:307 +0xb7
io.ReadFull(0x32d2ac0, 0xc427408300, 0xc426184118, 0x9, 0x9, 0x0, 0x1c20a50, 0x7fea24014f78)
	/usr/local/go/src/io/io.go:325 +0x73
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.readFrameHeader(0xc426184118, 0x9, 0x9, 0x32d2ac0, 0xc427408300, 0x0, 0xc400000000, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:237 +0xa3
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.(*Framer).ReadFrame(0xc4261840e0, 0x0, 0x32d31c0, 0xc420074040, 0xc42939cf00)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:492 +0x118
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*framer).readFrame(0xc4268ee750, 0xc4273cf2c0, 0xc4273cf2c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http_util.go:508 +0x53
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams(0xc4261842a0, 0xc4268ee8a0, 0x248fba0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:307 +0x3c4
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams(0xc42084ce00, 0x32f2040, 0xc4261842a0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:505 +0x1e8
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveHTTP2Transport(0xc42084ce00, 0x32f2b20, 0xc42403fc00, 0x32daac0, 0xc42031def0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:486 +0x5a6
github.com/cockroachdb/cockroach/vendor/g
```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_split_test.go
+++ b/pkg/storage/client_split_test.go
@@ -1925,3 +1925,118 @@ func TestUnsplittableRange(t *testing.T) {
 			origMaxBytes, repl.GetMaxBytes())
 	})
 }
+
+// TestPushTxnQueueDependencyCycleWithRangeSplit verifies that a range
+// split which occurs while a dependency cycle is partially underway
+// will cause the pending push txns to be retried such that they
+// relocate to the appropriate new range.
+func TestPushTxnQueueDependencyCycleWithRangeSplit(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	for _, read2ndPass := range []bool{false, true} {
+		t.Run(fmt.Sprintf("read-2nd-pass:%t", read2ndPass), func(t *testing.T) {
+			var pushCount int32
+			firstPush := make(chan struct{})
+
+			storeCfg := storage.TestStoreConfig(nil)
+			storeCfg.TestingKnobs.DisableSplitQueue = true
+			storeCfg.TestingKnobs.TestingEvalFilter =
+				func(filterArgs storagebase.FilterArgs) *roachpb.Error {
+					if _, ok := filterArgs.Req.(*roachpb.PushTxnRequest); ok {
+						if atomic.AddInt32(&pushCount, 1) == 1 {
+							close(firstPush)
+						}
+					}
+					return nil
+				}
+			stopper := stop.NewStopper()
+			defer stopper.Stop()
+			store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+			lhsKey := roachpb.Key("a")
+			rhsKey := roachpb.Key("b")
+
+			// Split at "a".
+			args := adminSplitArgs(lhsKey, lhsKey)
+			if _, pErr := client.SendWrapped(context.Background(), rg1(store), args); pErr != nil {
+				t.Fatalf("split at %q: %s", lhsKey, pErr)
+			}
+			lhs := store.LookupReplica(roachpb.RKey("a"), nil)
+
+			var txnACount, txnBCount int32
+
+			txnAWritesA := make(chan struct{})
+			txnAProceeds := make(chan struct{})
+			txnBWritesB := make(chan struct{})
+			txnBProceeds := make(chan struct{})
+
+			// Start txn to write key a.
+			txnACh := make(chan error)
+			go func() {
+				txnACh <- store.DB().Txn(context.Background(), func(ctx context.Context, txn *client.Txn) error {
+					if err := txn.Put(ctx, lhsKey, "value"); err != nil {
+						return err
+					}
+					if atomic.LoadInt32(&txnACount) == 0 {
+						close(txnAWritesA)
+						<-txnAProceeds
+					}
+					atomic.AddInt32(&txnACount, 1)
+					return txn.Put(ctx, rhsKey, "value-from-A")
+				})
+			}()
+			<-txnAWritesA
+
+			// Start txn to write key b.
+			txnBCh := make(chan error)
+			go func() {
+				txnBCh <- store.DB().Txn(context.Background(), func(ctx context.Context, txn *client.Txn) error {
+					if err := txn.Put(ctx, rhsKey, "value"); err != nil {
+						return err
+					}
+					if atomic.LoadInt32(&txnBCount) == 0 {
+						close(txnBWritesB)
+						<-txnBProceeds
+					}
+					atomic.AddInt32(&txnBCount, 1)
+					// Read instead of write key "a" if directed. This caused a
+					// PUSH_TIMESTAMP to be issued from txn B instead of PUSH_ABORT.
+					if read2ndPass {
+						if _, err := txn.Get(ctx, lhsKey); err != nil {
+							return err
+						}
+					} else {
+						if err := txn.Put(ctx, lhsKey, "value-from-B"); err != nil {
+							return err
+						}
+					}
+					return nil
+				})
+			}()
+			<-txnBWritesB
+
+			// Now, let txnA proceed before splitting.
+			close(txnAProceeds)
+			// Wait for the push to occur.
+			<-firstPush
+
+			// Split at "b".
+			args = adminSplitArgs(rhsKey, rhsKey)
+			if _, pErr := client.SendWrappedWith(context.Background(), store, roachpb.Header{
+				RangeID: lhs.RangeID,
+			}, args); pErr != nil {
+				t.Fatalf("split at %q: %s", rhsKey, pErr)
+			}
+
+			// Now that we've split, allow txnB to proceed.
+			close(txnBProceeds)
+
+			// Verify that both complete.
+			for i, ch := range []chan error{txnACh, txnBCh} {
+				if err := <-ch; err != nil {
+					t.Fatalf("%d: txn failure: %v", i, err)
+				}
+			}
+		})
+	}
+}

--- a/pkg/storage/push_txn_queue.go
+++ b/pkg/storage/push_txn_queue.go
@@ -91,6 +91,10 @@ type pendingTxn struct {
 	waiters []*waitingPush
 }
 
+func (pt *pendingTxn) getTxn() *roachpb.Transaction {
+	return pt.txn.Load().(*roachpb.Transaction)
+}
+
 // A pushTxnQueue enqueues PushTxn requests which are waiting on
 // extant txns with conflicting intents to abort or commit.
 //
@@ -123,21 +127,31 @@ func (ptq *pushTxnQueue) Enable() {
 	}
 }
 
-// ClearAndDisable empties the queue and returns all waiters. This
-// method should be invoked when the replica loses or transfers its
-// lease. Once invoked, no transaction may be enqueued or waiting
-// pushers added. Call Enable() when the lease is again acquired by
-// the replica.
-func (ptq *pushTxnQueue) ClearAndDisable() {
+// Clear empties the queue and returns all waiters. This method should
+// be invoked when the replica loses or transfers its lease. If
+// `disable` is true, future transactions may not be enqueued or
+// waiting pushers added. Call Enable() once the lease is again
+// acquired by the replica.
+func (ptq *pushTxnQueue) Clear(disable bool) {
 	ptq.mu.Lock()
 	var allWaiters []chan *roachpb.Transaction
 	for _, pt := range ptq.mu.txns {
 		for _, w := range pt.waiters {
 			allWaiters = append(allWaiters, w.pending)
 		}
+		if log.V(1) {
+			log.Infof(
+				context.Background(),
+				"clearing %d waiters for %s",
+				len(pt.waiters),
+				pt.getTxn().ID.Short(),
+			)
+		}
 		pt.waiters = nil
 	}
-	ptq.mu.txns = nil
+	if disable {
+		ptq.mu.txns = nil
+	}
 	ptq.mu.Unlock()
 
 	// Send on the pending waiter channels outside of the mutex lock.
@@ -194,6 +208,9 @@ func (ptq *pushTxnQueue) UpdateTxn(txn *roachpb.Transaction) {
 	delete(ptq.mu.txns, *txn.ID)
 	ptq.mu.Unlock()
 
+	if log.V(1) {
+		log.Infof(context.Background(), "updating %d waiters for %s", len(waiters), txn.ID.Short())
+	}
 	// Send on pending waiter channels outside of the mutex lock.
 	for _, w := range waiters {
 		w.pending <- txn
@@ -245,15 +262,19 @@ var errDeadlock = roachpb.NewErrorf("deadlock detected")
 // In the event of a dependency cycle of pushers leading to deadlock,
 // this method will return an errDeadlock error.
 func (ptq *pushTxnQueue) MaybeWait(
-	ctx context.Context, req *roachpb.PushTxnRequest,
+	ctx context.Context, repl *Replica, req *roachpb.PushTxnRequest,
 ) (*roachpb.PushTxnResponse, *roachpb.Error) {
 	if shouldPushImmediately(req) {
 		return nil, nil
 	}
 
 	ptq.mu.Lock()
-	if ptq.mu.txns == nil {
-		// Not enabled; do nothing.
+	// If the push txn queue is not enabled or if the request is not
+	// contained within the replica, do nothing. The request can fall
+	// outside of the replica after a split or merge. Note that the
+	// ContainsKey check is done under the push txn queue's lock to
+	// ensure that it's not cleared before an incorrect insertion happens.
+	if ptq.mu.txns == nil || !repl.ContainsKey(req.Key) {
 		ptq.mu.Unlock()
 		return nil, nil
 	}
@@ -265,7 +286,7 @@ func (ptq *pushTxnQueue) MaybeWait(
 		ptq.mu.Unlock()
 		return nil, nil
 	}
-	if txn := pending.txn.Load().(*roachpb.Transaction); isPushed(req, txn) {
+	if txn := pending.getTxn(); isPushed(req, txn) {
 		ptq.mu.Unlock()
 		return createPushTxnResponse(txn), nil
 	}
@@ -275,7 +296,19 @@ func (ptq *pushTxnQueue) MaybeWait(
 		pending: make(chan *roachpb.Transaction, 1),
 	}
 	pending.waiters = append(pending.waiters, push)
-
+	if log.V(2) {
+		if req.PusherTxn.ID != nil {
+			log.Infof(
+				ctx,
+				"%s pushing %s (%d pending)",
+				req.PusherTxn.ID.Short(),
+				req.PusheeTxn.ID.Short(),
+				len(pending.waiters),
+			)
+		} else {
+			log.Infof(ctx, "pushing %s (%d pending)", req.PusheeTxn.ID.Short(), len(pending.waiters))
+		}
+	}
 	ptq.mu.Unlock()
 
 	// Periodically refresh the pusher and pushee txns (with an
@@ -284,10 +317,6 @@ func (ptq *pushTxnQueue) MaybeWait(
 	// transactions.
 	r := retry.Start(ptq.store.cfg.RangeRetryOptions)
 	queryCh := r.NextCh()
-
-	// Keep local values for the pusher and pushee priorities to avoid
-	// modifying values in the original request on QueryTxn updates.
-	pusheePriority := req.PusheeTxn.Priority
 
 	for {
 		select {
@@ -317,15 +346,17 @@ func (ptq *pushTxnQueue) MaybeWait(
 			updatedPushee, _, pErr := ptq.queryTxnStatus(ctx, req.PusheeTxn, ptq.store.Clock().Now())
 			if pErr != nil {
 				return nil, pErr
-			} else if updatedPushee != nil {
-				pusheePriority = updatedPushee.Priority
-				pending.txn.Store(updatedPushee)
-				if isExpired(ptq.store.Clock().Now(), updatedPushee) {
-					if log.V(1) {
-						log.Warningf(ctx, "%s pushing expired txn %s", req.PusherTxn.ID.Short(), req.PusheeTxn.ID.Short())
-					}
-					return nil, nil
+			} else if updatedPushee == nil {
+				// Continue with push.
+				return nil, nil
+			}
+			pusheePriority := updatedPushee.Priority
+			pending.txn.Store(updatedPushee)
+			if isExpired(ptq.store.Clock().Now(), updatedPushee) {
+				if log.V(1) {
+					log.Warningf(ctx, "pushing expired txn %s", req.PusheeTxn.ID.Short())
 				}
+				return nil, nil
 			}
 
 			if req.PusherTxn.ID != nil {
@@ -355,7 +386,15 @@ func (ptq *pushTxnQueue) MaybeWait(
 						dependents = append(dependents, id.Short())
 					}
 					if log.V(2) {
-						log.Infof(ctx, "%s has dependencies=%s", req.PusherTxn.ID.Short(), dependents)
+						log.Infof(
+							ctx,
+							"%s (%d), pushing %s (%d), has dependencies=%s",
+							req.PusherTxn.ID.Short(),
+							pusherPriority,
+							req.PusheeTxn.ID.Short(),
+							pusheePriority,
+							dependents,
+						)
 					}
 					push.mu.Unlock()
 
@@ -364,7 +403,7 @@ func (ptq *pushTxnQueue) MaybeWait(
 						p1, p2 := pusheePriority, pusherPriority
 						if p1 < p2 || (p1 == p2 && bytes.Compare(req.PusheeTxn.ID.GetBytes(), req.PusherTxn.ID.GetBytes()) < 0) {
 							if log.V(1) {
-								log.Warningf(
+								log.Infof(
 									ctx,
 									"%s breaking deadlock by force push of %s; dependencies=%s",
 									req.PusherTxn.ID.Short(),

--- a/pkg/storage/push_txn_queue_test.go
+++ b/pkg/storage/push_txn_queue_test.go
@@ -183,7 +183,7 @@ func TestPushTxnQueueEnableDisable(t *testing.T) {
 
 	retCh := make(chan RespWithErr, 1)
 	go func() {
-		resp, pErr := ptq.MaybeWait(context.Background(), &req)
+		resp, pErr := ptq.MaybeWait(context.Background(), tc.repl, &req)
 		retCh <- RespWithErr{resp, pErr}
 	}()
 
@@ -196,7 +196,7 @@ func TestPushTxnQueueEnableDisable(t *testing.T) {
 	})
 
 	// Now disable the queue and make sure the waiter is returned.
-	ptq.ClearAndDisable()
+	ptq.Clear(true /* disable */)
 	if ptq.isEnabled() {
 		t.Errorf("expected queue to be disabled")
 	}
@@ -225,7 +225,7 @@ func TestPushTxnQueueEnableDisable(t *testing.T) {
 	}
 	ptq.mu.Unlock()
 
-	if resp, pErr := ptq.MaybeWait(context.TODO(), &req); resp != nil || pErr != nil {
+	if resp, pErr := ptq.MaybeWait(context.TODO(), tc.repl, &req); resp != nil || pErr != nil {
 		t.Errorf("expected nil resp and err as queue is disabled; got %+v, %s", resp, pErr)
 	}
 }
@@ -255,7 +255,7 @@ func TestPushTxnQueueCancel(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	retCh := make(chan RespWithErr, 1)
 	go func() {
-		resp, pErr := ptq.MaybeWait(ctx, &req)
+		resp, pErr := ptq.MaybeWait(ctx, tc.repl, &req)
 		retCh <- RespWithErr{resp, pErr}
 	}()
 
@@ -306,7 +306,7 @@ func TestPushTxnQueueUpdateTxn(t *testing.T) {
 
 	retCh := make(chan RespWithErr, 2)
 	go func() {
-		resp, pErr := ptq.MaybeWait(context.Background(), &req1)
+		resp, pErr := ptq.MaybeWait(context.Background(), tc.repl, &req1)
 		retCh <- RespWithErr{resp, pErr}
 	}()
 	testutils.SucceedsSoon(t, func() error {
@@ -318,7 +318,7 @@ func TestPushTxnQueueUpdateTxn(t *testing.T) {
 	})
 
 	go func() {
-		resp, pErr := ptq.MaybeWait(context.Background(), &req2)
+		resp, pErr := ptq.MaybeWait(context.Background(), tc.repl, &req2)
 		retCh <- RespWithErr{resp, pErr}
 	}()
 	testutils.SucceedsSoon(t, func() error {
@@ -371,7 +371,7 @@ func TestPushTxnQueueUpdateNotPushedTxn(t *testing.T) {
 
 	retCh := make(chan RespWithErr, 1)
 	go func() {
-		resp, pErr := ptq.MaybeWait(context.Background(), &req)
+		resp, pErr := ptq.MaybeWait(context.Background(), tc.repl, &req)
 		retCh <- RespWithErr{resp, pErr}
 	}()
 
@@ -443,7 +443,7 @@ func TestPushTxnQueuePusheeExpires(t *testing.T) {
 
 	retCh := make(chan RespWithErr, 2)
 	go func() {
-		resp, pErr := ptq.MaybeWait(context.Background(), &req1)
+		resp, pErr := ptq.MaybeWait(context.Background(), tc.repl, &req1)
 		retCh <- RespWithErr{resp, pErr}
 	}()
 	testutils.SucceedsSoon(t, func() error {
@@ -455,7 +455,7 @@ func TestPushTxnQueuePusheeExpires(t *testing.T) {
 	})
 
 	go func() {
-		resp, pErr := ptq.MaybeWait(context.Background(), &req2)
+		resp, pErr := ptq.MaybeWait(context.Background(), tc.repl, &req2)
 		retCh <- RespWithErr{resp, pErr}
 	}()
 	testutils.SucceedsSoon(t, func() error {
@@ -507,7 +507,7 @@ func TestPushTxnQueuePusherUpdate(t *testing.T) {
 
 	retCh := make(chan RespWithErr, 1)
 	go func() {
-		resp, pErr := ptq.MaybeWait(context.Background(), &req)
+		resp, pErr := ptq.MaybeWait(context.Background(), tc.repl, &req)
 		retCh <- RespWithErr{resp, pErr}
 	}()
 
@@ -585,7 +585,7 @@ func TestPushTxnQueueDependencyCycle(t *testing.T) {
 	retCh := make(chan RespWithErr, 3)
 	for _, req := range []*roachpb.PushTxnRequest{reqA, reqB, reqC} {
 		go func(req *roachpb.PushTxnRequest) {
-			resp, pErr := ptq.MaybeWait(ctx, req)
+			resp, pErr := ptq.MaybeWait(ctx, tc.repl, req)
 			retCh <- RespWithErr{resp, pErr}
 		}(req)
 	}
@@ -661,7 +661,7 @@ func TestPushTxnQueueDependencyCycleWithPriorityInversion(t *testing.T) {
 	retCh := make(chan ReqWithErr, 2)
 	for _, req := range []*roachpb.PushTxnRequest{reqA, reqB} {
 		go func(req *roachpb.PushTxnRequest) {
-			_, pErr := ptq.MaybeWait(ctx, req)
+			_, pErr := ptq.MaybeWait(ctx, tc.repl, req)
 			retCh <- ReqWithErr{req, pErr}
 		}(req)
 	}

--- a/pkg/storage/replica_command.go
+++ b/pkg/storage/replica_command.go
@@ -1554,7 +1554,9 @@ func evalPushTxn(
 		reply.PusheeTxn.Timestamp = args.Now // see method comment
 		// Setting OrigTimestamp bumps LastActive(); see #9265.
 		reply.PusheeTxn.OrigTimestamp = args.Now
-		return EvalResult{}, engine.MVCCPutProto(ctx, batch, cArgs.Stats, key, hlc.Timestamp{}, nil, &reply.PusheeTxn)
+		result := EvalResult{}
+		result.Local.updatedTxn = &reply.PusheeTxn
+		return result, engine.MVCCPutProto(ctx, batch, cArgs.Stats, key, hlc.Timestamp{}, nil, &reply.PusheeTxn)
 	}
 	// Start with the persisted transaction record as final transaction.
 	reply.PusheeTxn = existTxn.Clone()
@@ -1600,7 +1602,7 @@ func evalPushTxn(
 		reason = "pusher has priority"
 		pusherWins = true
 	case args.Force:
-		reason = "forced txn abort"
+		reason = "forced push"
 		pusherWins = true
 	}
 

--- a/pkg/storage/replica_proposal.go
+++ b/pkg/storage/replica_proposal.go
@@ -424,7 +424,7 @@ func (r *Replica) leasePostApply(
 	if leaseChangingHands && !iAmTheLeaseHolder {
 		// Also clear and disable the push transaction queue. Any waiters
 		// must be redirected to the new lease holder.
-		r.pushTxnQueue.ClearAndDisable()
+		r.pushTxnQueue.Clear(true /* disable */)
 	}
 
 	if !iAmTheLeaseHolder && r.IsLeaseValid(newLease, r.store.Clock().Now()) {

--- a/pkg/storage/store.go
+++ b/pkg/storage/store.go
@@ -1915,6 +1915,12 @@ func (s *Store) SplitRange(ctx context.Context, origRng, newRng *Replica) error 
 	copyDesc.EndKey = append([]byte(nil), newDesc.StartKey...)
 	origRng.setDescWithoutProcessUpdate(&copyDesc)
 
+	// Clear the LHS push txn queue, to redirect to the RHS if
+	// appropriate. We do this after setDescWithoutProcessUpdate
+	// to ensure that no pre-split commands are inserted into the
+	// pushTxnQueue after we clear it.
+	origRng.pushTxnQueue.Clear(false /* disable */)
+
 	if kr := s.mu.replicasByKey.ReplaceOrInsert(origRng); kr != nil {
 		return errors.Errorf("replicasByKey unexpectedly contains %s when inserting replica %s", kr, origRng)
 	}
@@ -1980,6 +1986,10 @@ func (s *Store) MergeRange(
 	if err := s.removeReplicaImpl(ctx, subsumedRng, *subsumedDesc, false); err != nil {
 		return errors.Errorf("cannot remove range %s", err)
 	}
+
+	// Clear the RHS push txn queue, to redirect to the LHS if
+	// appropriate.
+	subsumedRng.pushTxnQueue.Clear(false /* disable */)
 
 	// Update the end key of the subsuming range.
 	copy := *subsumingDesc
@@ -2521,7 +2531,7 @@ func (s *Store) Send(
 		// txn response or else allow this request to proceed.
 		if ba.IsSinglePushTxnRequest() {
 			pushReq := ba.Requests[0].GetInner().(*roachpb.PushTxnRequest)
-			pushResp, pErr := repl.pushTxnQueue.MaybeWait(ctx, pushReq)
+			pushResp, pErr := repl.pushTxnQueue.MaybeWait(repl.AnnotateCtx(ctx), repl, pushReq)
 			// Copy the request in anticipation of setting the force arg and
 			// updating the Now timestamp (see below).
 			pushReqCopy := *pushReq
@@ -2540,7 +2550,7 @@ func (s *Store) Send(
 			// request may have been waiting to push the txn. If we don't
 			// move the timestamp forward to the current time, we may fail
 			// to push a txn which has expired.
-			pushReqCopy.Now = s.Clock().Now()
+			pushReqCopy.Now.Forward(s.Clock().Now())
 			ba.Requests = nil
 			ba.Add(&pushReqCopy)
 		}

